parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g

```diff
@@ -479,6 +479,8 @@ alterStatementSuffixExecute
     -> ^(TOK_ALTERTABLE_EXECUTE KW_EXPIRE_SNAPSHOTS $expireParam)
     | KW_EXECUTE KW_SET_CURRENT_SNAPSHOT LPAREN (snapshotParam=Number) RPAREN
     -> ^(TOK_ALTERTABLE_EXECUTE KW_SET_CURRENT_SNAPSHOT $snapshotParam)
+    | KW_EXECUTE KW_FASTFORWARD sourceBranch=StringLiteral KW_TO targetBranch=StringLiteral
+    -> ^(TOK_ALTERTABLE_EXECUTE KW_FASTFORWARD $sourceBranch $targetBranch)
```

Comment:
could we make targetBranch optional and if not specified fastforward to master branch

Reply:
Also, i don't think we need KW_TO in a syntax

Reply:
````
alter table ice01 execute fast-forward test1;
alter table ice01 execute fast-forward main test1;
````

Reply:
btw, why do we need `execute`? it doesn't add any syntactic value

Reply:
We have every function calls of iceberg under execute, rollback, set current snapshot, expire snapshot. they also followed same syntax rather than having alter table ice01 expire snapshot 10;

So, wanted to stay in the same area. And we have couple of more coming, so they all can stay in same space

I will make the target branch as optional. 

Regarding TO.
```
alter table ice01 fast-forward test1 to main;
```

Doesn't this not make more sense like we are fast forwarding the test1 branch to where the main branch is?

Reply:
1. you are  fast-forwarding  test1 to the head of main branch. it should be reversed.
2. source could be optional. alter table execute ice01 fast-forward test1   
fast-forwards  main to the head of test1 branch

Reply:
I have added new tests and made changes.
it cases
it has ff main to test2
just ff test1 and it fast forwards main to test1
ff test01 to test1

Let me know if it works

Reply:
It feels a little strange omitting `KW_TO` keyword. IMO, I think it is more reasonable to retain the `KW_TO` keyword.
Spark fast_fowrward procedure also has two syntax, one of is has `KW_TO` keyword.

https://github.com/apache/iceberg/blob/1f8dbae31aab98654ebf4d64d37b04bff8eb22fd/spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestFastForwardBranchProcedure.java#L123-L124

https://github.com/apache/iceberg/blob/master/spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestFastForwardBranchProcedure.java#L81-L82

Reply:
@zhangbutao why do we need the 'TO' keyword?
````
git merge --ff-only master feature
````

Reply:
I see you. If we want to refer to git syntax, i am ok with this change.
I just thought `TO` keyword would make syntax more smoothly, but the thought maybe a little subjectively.
Anyway, i am ok with or without `TO`.

Reply:
"CALL %s.system.fast_forward(table => '%s', branch => '%s', to => '%s')" isn't that just a named params, => in scala is used to declare a lambda